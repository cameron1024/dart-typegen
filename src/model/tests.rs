use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
