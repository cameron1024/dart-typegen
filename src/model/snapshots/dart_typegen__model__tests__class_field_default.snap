---
source: src/model/tests.rs
expression: ron
---
Library {
    preamble: None,
    postamble: None,
    defaults: None,
    enums: [],
    classes: [
        Class {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Foo",
                span: SourceSpan {
                    offset: SourceOffset(
                        57,
                    ),
                    length: 5,
                },
            },
            fields: [
                Field {
                    name: SpannedScalar {
                        value: "x",
                        span: SourceSpan {
                            offset: SourceOffset(
                                73,
                            ),
                            length: 3,
                        },
                    },
                    ty: SpannedScalar {
                        value: "String",
                        span: SourceSpan {
                            offset: SourceOffset(
                                82,
                            ),
                            length: 8,
                        },
                    },
                    defaults_to: Some(
                        Value {
                            type_name: None,
                            literal: Spanned {
                                span: Span(
                                    109,
                                    116,
                                ),
                                value: String(
                                    "hello",
                                ),
                            },
                        },
                    ),
                    defaults_to_dart: None,
                    docs: None,
                },
                Field {
                    name: SpannedScalar {
                        value: "y",
                        span: SourceSpan {
                            offset: SourceOffset(
                                130,
                            ),
                            length: 3,
                        },
                    },
                    ty: SpannedScalar {
                        value: "int",
                        span: SourceSpan {
                            offset: SourceOffset(
                                139,
                            ),
                            length: 5,
                        },
                    },
                    defaults_to: None,
                    defaults_to_dart: Some(
                        SpannedScalar {
                            value: "123 + 234",
                            span: SourceSpan {
                                offset: SourceOffset(
                                    168,
                                ),
                                length: 11,
                            },
                        },
                    ),
                    docs: None,
                },
            ],
            docs: None,
            extra_dart: [],
        },
    ],
    unions: [],
}
